extensions/amp-social-share/1.0/amp-social-share.js

```diff
@@ -100,6 +145,19 @@ class AmpSocialShare extends PreactBaseElement {
     });
   }
 
+  /** @override */
+  mutationObserverCallback(mutations) {
+    const typeConfig = updateTypeConfig(this.element, mutations);
+    if (typeConfig) {
+      this.renderWithHrefAndTarget_(typeConfig);
+    }
+  }
+
+  /** @override */
+  getAdditionalMutationObserverInitProperties() {
+    return {attributeOldValue: true};
```

Comment:
I see your logic, but is there any way to express this w/o needing this option? It's a significant slow down on mutation  observer API since it reduces the extent to which batching can happen. TBH, it's a lot cheaper to keep private state on the instance level here and do a diff then read old values via the mutation observer.

Reply:
Updated in latest to track state within component.